src/raw.rs

```diff
@@ -117,7 +118,7 @@ pub struct RawValue {
 
 impl RawValue {
     fn from_borrowed(json: &str) -> &Self {
-        unsafe { mem::transmute::<&str, &RawValue>(json) }
+        unsafe { &*(json as *const str as *const RawValue) }
```

Comment:
Had to change this to get clippy to compile.

Reply:
I would prefer not to make this change. I think this lint is bad and I have suppressed it in https://github.com/serde-rs/json/commit/88ee1a69dab73a45cd0eca5db505164fcd48cbf1. The original code is clearer and safer.